Java/core/src/main/java/com/amazon/randomcutforest/tree/BoundingBox.java

```diff
@@ -27,7 +27,7 @@
  * always the smallest BoundingBox that contains all leaf points which are
  * descendents of the Node.
  */
-public class BoundingBox {
+public class BoundingBox implements IBox<double[]> {
```

Comment:
I think we should either change this class name to Box or else change the interface name to IBoundingBox.